src/main/java/geocode/kdtree/KDTree.java

```diff
@@ -44,7 +44,22 @@ public KDTree( List<T> items ) {
     }
 
     public T findNearest( T search ) {
-        return findNearest(root, search, 0).location;
+        return findNearest(search, null);
+    }
+
+    /**
+     * Finds nearest point within a max distance (in km)
+     */
+    public T findNearest(T search, Double maxDistance) {
+        T nearest = findNearest(root, search, 0).location;
+        if (null != maxDistance) {
+            int earthRadiusInKm = 6371;
+            double distanceInKm = Math.sqrt(nearest.squaredDistance(search)) * earthRadiusInKm;
+            if (distanceInKm > maxDistance) {
+                nearest = null;
```

Comment:
Is this functionality just for part of our business requirements?

Reply:
Yes, we didn't want it to return the nearest location if it was over some maximum distance away. It should return no postal code instead.